request-invoice/requestinvoice.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+
+import multiprocessing
+
+from flask import Flask
+from flask import jsonify
+from flask_limiter import Limiter
+from flask_limiter.util import get_remote_address
+from tornado.wsgi import WSGIContainer
+from tornado.httpserver import HTTPServer
+from tornado.ioloop import IOLoop
+
+from pyln.client import LightningRpc, Plugin
+from time import time
+from random import random
+
+
+# change this
+secret= 'caba27ba-45c7-4495-aa53-fd6a5866fbd8'
+
+
+plugin = Plugin()
+app = Flask(__name__)
+limiter = Limiter(
+    app,
+    key_func=get_remote_address,
+    default_limits=["200 per day", "50 per hour"]
+)
+
+
+@limiter.limit("20 per minute")
+def getinvoice(amount, description):
+        global plugin
+        label = "ln-getinvoice-{}".format(random())
+        invoice = plugin.rpc.invoice(int(amount)*1000, label, description)
+        return invoice
+
+
+def worker(port):
+    app.config['SECRET_KEY'] = secret
+    app.add_url_rule('/invoice/<int:amount>/<description>', 'getinvoice', getinvoice)
+    http_server = HTTPServer(WSGIContainer(app))
+    http_server.listen(port)
+    IOLoop.instance().start()
+    return
+
+
+jobs = {}
+
+
+def start_server(port):
+    if port in jobs:
+        return False, "server already running"
+
+    p = multiprocessing.Process(
+        target=worker, args=[port], name="server on port {}".format(port))
+    p.daemon = True
```

Comment:
Running in a separate process might make future things harder since we don't really have a way to communicate with the subprocess at all. I'd suggest using `threading.Thread()` instead.